### PR TITLE
feat: add `enable` to chain-monitor config

### PR DESCRIPTION
**Pull request by mask-pp**
### Purpose or design rationale of this PR

*Describe your change. Make sure to answer these three questions: What does this PR do? Why does it do it? How does it do it?*

Open chain-monitor and let it optional.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [x] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [ ] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [ ] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [ ] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: rollup/internal/controller/relayer/l2_relayer.go [hunk 1]
```diff
@@ -553,6 +554,9 @@ type batchStatusResponse struct {
 }
 
 func (r *Layer2Relayer) getBatchStatusByIndex(batchIndex uint64) (bool, error) {
+	if r.chainMonitorClient == nil {
+		return true, nil
+	}
```
Comment (georgehao) [inline rollup/internal/controller/relayer/l2_relayer.go]: how about move this check logic out of here , to the call place? L437?

**Review by georgehao — DISMISSED** [review]

File: rollup/internal/controller/relayer/l2_relayer.go (hunk 1 above)
Comment (mask-pp) [reply rollup/internal/controller/relayer/l2_relayer.go]: Agree and changed at https://github.com/scroll-tech/scroll/pull/942/commits/57b776070dc2b37528fdb6a950bca47d689b3280.

File: rollup/internal/controller/relayer/l2_relayer.go [hunk 2]
```diff
@@ -433,17 +434,17 @@ func (r *Layer2Relayer) ProcessCommittedBatches() {
...
-		//	r.metrics.rollupL2ChainMonitorLatestFailedCall.Inc()
-		//	log.Warn("failed to get batch status, please check chain_monitor api server", "batch_index", batch.Index, "err", err)
-		//	return
-		//}
-		//if !batchStatus {
-		//	r.metrics.rollupL2ChainMonitorLatestFailedBatchStatus.Inc()
-		//	log.Error("the batch status is not right, stop finalize batch and check the reason", "batch_index", batch.Index)
-		//	return
-		//}
+		batchStatus, err := r.getBatchStatusByIndex(batch.Index)
```
Comment (georgehao) [inline rollup/internal/controller/relayer/l2_relayer.go]: I think the checker `if !r.cfg.ChainMonitor.EnableChainMonitor {` put here will be more suitable

Comment (mask-pp) [reply rollup/internal/controller/relayer/l2_relayer.go]: fixed.

**Review by georgehao — DISMISSED** [review]

**Review by Thegaram — DISMISSED** [review]

File: rollup/internal/config/relayer.go [hunk 3]
```diff
@@ -39,9 +39,10 @@ type SenderConfig struct {
 
 // ChainMonitor this config is used to get batch status from chain_monitor API.
 type ChainMonitor struct {
-	TimeOut  int    `json:"timeout"`
-	TryTimes int    `json:"try_times"`
-	BaseURL  string `json:"base_url"`
+	EnableChainMonitor bool   `json:"enable_chain_monitor"`
```
Comment (0xmountaintop) [inline rollup/internal/config/relayer.go]:
```suggestion
	Enable bool   `json:"enable"`
```

**Review by colinlyguo — DISMISSED** [review]

**Review by 0xmountaintop — APPROVED** [review]

**Review by colinlyguo — APPROVED** [review]

**Pull request merged**